src/happyapi/oauth2/auth.clj

```diff
@@ -145,8 +145,19 @@
   (boolean (or refresh_token private_key)))
 
 (defn credential-scopes [credentials]
-  (set (some-> (:scope credentials) (str/split #" "))))
-
-(defn has-scopes? [credentials scopes]
-  ;; TODO: scopes have a hierarchy
-  (set/subset? (set scopes) (credential-scopes credentials)))
+  (set (some-> credentials
```

Comment:
I usually prefer splitting threading macros up to different lines that are left-aligned and can be read from top to bottom -- just as a chain of pipes and filters.

If that's bad style for some reason (e.g. a style convention), let's fix it. I saw the rest of the code has a slightly different prevailing style.